bgpd/bgp_nexthop.c

```diff
@@ -763,15 +786,66 @@ static void bgp_show_nexthop_paths(struct vty *vty, struct bgp *bgp,
 			vty_out(vty, "    %d/%d %pBD %s flags 0x%x\n",
 				afi, safi, dest, bgp_path->name_pretty, path->flags);
 	}
+	if (json)
+		json_object_object_add(json, "paths", paths);
 }
 
 static void bgp_show_nexthops_detail(struct vty *vty, struct bgp *bgp,
-				     struct bgp_nexthop_cache *bnc)
+				     struct bgp_nexthop_cache *bnc,
+				     json_object *json)
 {
 	char buf[PREFIX2STR_BUFFER];
 	struct nexthop *nexthop;
 
 	for (nexthop = bnc->nexthop; nexthop; nexthop = nexthop->next) {
+		if (json) {
+			switch (nexthop->type) {
```

Comment:
There could be more than 1 "gate" for a nexthop. Can we make an array of gates and add each entry to this array?

For example:

spine-1# show bgp vrf all nexthop 

Instance default:
Current BGP nexthop cache:
 6.0.0.28 valid [IGP metric 0], #paths 40
  **gate** fe80::202:ff:fe00:1b, if swp1
  **gate** fe80::202:ff:fe00:1f, if swp2
  Last update: Wed Oct 26 00:11:51 2022

Reply:
Fixed